### PR TITLE
Added setSelected, toggleSelected, and getSelected to all charts that…

### PR DESCRIPTION
**Pull request by krahs1**
> Explain the **details** for making this change. What existing problem does the pull request solve?

The current Angular Chart components api does not allow selection functionality to the core components

> **Related issue (required)**:

 #100 

> **Steps necessary to review your pull request (required)**:

This is an api issue so no visible issues can be seen in the control.  The examples were expanded to include buttons to test that selection is working and mimic the core components.

> Other Details:

### CODE REVIEW

File: src/app/area/area.demo.html [hunk 1]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Area Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/area/area.demo.html]: File needs a new line at the end

File: src/app/bar-grouped/bar-grouped.demo.html [hunk 2]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Grouped Bar Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/bar-grouped/bar-grouped.demo.html]: File needs a new line at the end

File: src/soho/pie/soho-pie.d.ts [hunk 3]
```diff
@@ -89,26 +89,31 @@ interface SohoPieSelectEvent {
...
+interface SohoPieSelected {
+  fieldName: string;
+  fieldValue: any;
+}
+
 interface JQuery {
   pie(options?: SohoPieOptions): JQuery;
-}
\ No newline at end of file
+}
```
Comment (fitzorama) [inline src/soho/pie/soho-pie.d.ts]: File needs a new line at the end

File: src/soho/line/soho-line.d.ts [hunk 4]
```diff
@@ -70,25 +70,31 @@ interface SohoLineSelectEvent {
...
+  groupIndex: number;
+  fieldName: string;
+  fieldValue: any;
 }
 
 interface JQuery {
   line(options?: SohoLineOptions): JQuery;
-}
\ No newline at end of file
+}
```
Comment (fitzorama) [inline src/soho/line/soho-line.d.ts]: File needs a new line at the end. Looks like there are a lot of these.

File: src/app/pie/pie.demo.html [hunk 5]
```diff
@@ -13,5 +13,7 @@ <h2 class="widget-title">Pie Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/pie/pie.demo.html]: File needs a new line at the end

File: src/app/column/column.demo.html [hunk 6]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Column Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/column/column.demo.html]: File needs a new line at the end

File: src/app/column-stacked/column-stacked.demo.html [hunk 7]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Stacked Column Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/column-stacked/column-stacked.demo.html]: File needs a new line at the end

File: src/app/column-grouped/column-grouped.demo.html [hunk 8]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Grouped Column Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/column-grouped/column-grouped.demo.html]: File needs a new line at the end

File: src/app/bubble/bubble.demo.html [hunk 9]
```diff
@@ -12,4 +12,10 @@ <h2 class="widget-title">Bubble Chart Title</h2>
...
   </div>
-</div>
\ No newline at end of file
+</div>
+<div class="row">
+  <div class="twelve columns">
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
+  </div>
+</div>
```
Comment (fitzorama) [inline src/app/bubble/bubble.demo.html]: File needs a new line at the end

File: src/app/bar/bar.demo.html [hunk 10]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Bar Chart Title</h2>
         </div>
       </div>
     </div>
+    <button soho-button (click)="setChartSelection()">setSelected</button>
+    <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (fitzorama) [inline src/app/bar/bar.demo.html]: File needs a new line at the end

File: src/app/bar-stacked/bar-stacked.demo.html [hunk 11]
```diff
@@ -11,5 +11,7 @@ <h2 class="widget-title">Stacked Bar Chart Title</h2>
           </div>
         </div>
       </div>
+      <button soho-button (click)="setChartSelection()">setSelected</button>
+      <button soho-button (click)="toggleChartSelection()">toggleSelected</button>
     </div>
-  </div>
\ No newline at end of file
+  </div>
```
Comment (fitzorama) [inline src/app/bar-stacked/bar-stacked.demo.html]: File needs a new line at the end

File: src/soho/bar/soho-bar.component.ts [hunk 12]
```diff
@@ -225,4 +225,16 @@ export class SohoBarComponent implements AfterViewInit, OnDestroy {
     }
   }
 
+  public setSelected(selected: SohoBarSelected) {
+    this.bar.setSelected(selected);
+  }
```
Comment (pwpatton) [inline src/soho/bar/soho-bar.component.ts]: This is really what soho bar.js takes? a SohoBarSelect?

Comment (krahs1) [reply src/soho/bar/soho-bar.component.ts]: Each of the Charts type has its own definition file so it seemed that was the right way to go. Some chart take different params

**Pull request merged**